### ISSUE TITLE
i can't import class which start with org.apache.iceberg.relocated

### ISSUE THREAD

**Issue by fuchanghai** [apache/iceberg#3028]
![image](https://user-images.githubusercontent.com/33984497/130896898-c8d730c9-0580-4737-ad45-9968476aab70.png)

**Comment by openinx** [comment]
We've encountered many issues or mail list from the iceberg beginners,    I think it would be better if we could add a document in the https://iceberg.apache.org/community/ page.

**Comment by fuchanghai** [comment]
> We've encountered many issues or mail list from the iceberg beginners, I think it would be better if we could add a document in the https://iceberg.apache.org/community/ page.

It’s the first time I use gradle, maybe I have to understand gradle first to solve the environment dependency problem。
I'm sorry to bother you because of some low-level questions

**Comment by RussellSpitzer** [comment]
Try running the build, usually after building the project Intellij will be able to find the shaded libs. The issues is that the build puts those files in the proper place, so intellij doesn't think they are real until you've run the build.

I agree with @openinx we should document this! I think we had a pr for it at one point ... @kbendick ?

**Comment by myandpr** [comment]
If after building the project the problem hasn`t been solved yet, you can try this:
```
./gradlew clean build -x test -x javadoc -x integrationTest
```
if Intellij still cant`t find the shaded libs, you can "Reload All Gradle Projects". Usually can solve the problem.  @fuchanghai
![image](https://user-images.githubusercontent.com/20159351/131436664-026699e3-10c2-40c8-bca3-bca619f04f76.png)

**Comment by fuchanghai** [comment]
> If after building the project the problem hasn`t been solved yet, you can try this:
> 
> ```
> ./gradlew clean build -x test -x javadoc -x integrationTest
> ```
> 
> if Intellij still cant`t find the shaded libs, you can "Reload All Gradle Projects". Usually can solve the problem. @fuchanghai
> ![image](https://user-images.githubusercontent.com/20159351/131436664-026699e3-10c2-40c8-bca3-bca619f04f76.png)

I ran this project on windows, so there are some line break problems when I build。(It needs '\n 'but gave '\r\n')
![image](https://user-images.githubusercontent.com/33984497/131437213-8bde219b-68fa-4e5f-b8ae-6972d0bbc6b2.png)

**Comment by fuchanghai** [comment]
It seems to work on my Mac, but not on Windows

**Comment by myandpr** [comment]
yes,obviously it`s file format error.Maybe your code is from linux os and you read it in win os.Please pull code from github directly.

**Comment by theoxu31** [comment]
I still faces the same issue, after applied all the tries mentioned above

**Comment by ahmedriza** [comment]
Even adding the bundled Guava manually to the Intellij classpath doesn't help.  After adding manually to the classpath in Intellij:

<img width="865" alt="image" src="https://github.com/apache/iceberg/assets/2463311/e4de51c5-b1f8-45cb-ac62-bbdbb53560ec">

We get this:

<img width="853" alt="image" src="https://github.com/apache/iceberg/assets/2463311/4d2505d4-dc14-47a2-bb9a-2a971f7e2e8b">

This is rather odd, since now it complains that it can find `Maps` but manages to resolve the package!

**Comment by xuqi1633** [comment]
After compiling the project, a relocated guava jar file will be generated under the bundled-guava module
```
./gradlew clean build -x test -x javadoc -x integrationTest
```
![image](https://github.com/apache/iceberg/assets/70441327/51c2d53f-c50a-45ce-91eb-c44f9e03d2bb)
![image](https://github.com/apache/iceberg/assets/70441327/b75e709c-2238-4a91-b6f6-68cf28ac1fd8)

If other modules also rely on the bundled-guava module, add the iceberg-bundled-guava-1.4.0-SNAPSHOT libraries to the module
![image](https://github.com/apache/iceberg/assets/70441327/54bbd4fa-5265-41e6-9a0a-15242c9225f5)

after adding libraries
![image](https://github.com/apache/iceberg/assets/70441327/3eaa6bf6-bc94-4cd4-89ac-99fabd6e356f)

@ahmedriza

**Comment by jiamin13579** [comment]
> I reslove this problem to revert IDEA version from 2023.2 to 2022.3. I think this is a bug for IDEA 2023 @ahmedriza @theoxu31

I also encountered the same problem

**Comment by ajantha-bhat** [comment]
This is the IDE issue.
https://youtrack.jetbrains.com/issue/IDEA-321592/IJ-misses-shadowjar-output-on-classpath

Will be fixed in 2023.3
